abci/src/signatures.rs

```diff
@@ -113,24 +133,62 @@ impl SignDigest for CanonicalVote {
     }
 }
 
-impl SignDigest for VoteExtension {
-    fn sign_digest(
+impl Signable for VoteExtension {
+    fn calculate_sign_hash(
         &self,
         chain_id: &str,
         quorum_type: u8,
         quorum_hash: &[u8; 32],
         height: i64,
         round: i32,
     ) -> Result<Vec<u8>, Error> {
-        let request_id = sign_request_id(VOTE_EXTENSION_REQUEST_ID_PREFIX, height, round);
-        let sign_bytes_hash = self.sha256(chain_id, height, round)?;
-
-        Ok(sign_digest(
+        let (request_id, sign_bytes_hash) = match self.r#type() {
+            VoteExtensionType::ThresholdRecover => {
+                let request_id = sign_request_id(VOTE_EXTENSION_REQUEST_ID_PREFIX, height, round);
+                let sign_bytes_hash = self.calculate_msg_hash(chain_id, height, round)?;
+
+                (request_id, sign_bytes_hash)
+            },
+
+            VoteExtensionType::ThresholdRecoverRaw => {
+                let mut sign_bytes_hash = self.extension.clone();
+                sign_bytes_hash.reverse();
+
+                let request_id = self.sign_request_id.clone().unwrap_or_default();
+                let request_id = if request_id.is_empty() {
+                    sign_request_id(VOTE_EXTENSION_REQUEST_ID_PREFIX, height, round)
+                } else {
+                    // we do double-sha256, and then reverse bytes
+                    let mut request_id = lhash::sha256(&lhash::sha256(&request_id));
+                    request_id.reverse();
+                    request_id.to_vec()
+                };
+
+                (request_id, sign_bytes_hash)
+            },
+
+            VoteExtensionType::Default => unimplemented!(
+                "vote extension of type {:?} cannot be signed",
+                self.r#type()
+            ),
+        };
+        let sign_hash = sign_hash(
             quorum_type,
             quorum_hash,
-            request_id[..].try_into().unwrap(),
+            request_id[..]
+                .try_into()
+                .expect("invalid request ID length"),
             &sign_bytes_hash,
-        ))
+        );
+
+        tracing::trace!(
```

Comment:
You are gonna remove it before merge right?

Reply:
added comment to remove once withdrawals are done